### DISCUSSION TITLE
Re-enable Flash Player

### DISCUSSION THREAD

**Discussion by bradut** [GerryDazoo/Slinger#465] [General]
In case anyone wants to use SlingPlayer 2.0.4, they need to have a working instance of Adobe Flash on their machine.
Unfortunately Flash has been disabled by Microsoft in 2021, and, to get it back, you may use a project on GitHub that disables the time bomb:
https://gist.github.com/KuromeSan/56d8b724c0696b54f9f81994ae3591d1

(I see that in the meantime the author created a new repository: https://github.com/KuromeSan/FlashPatcher)

The app that diffuses the time bomb assumes that your computer has had Flash Player installed at a certain time, and currently is only disabled, therefore you did not removed it.
In case you have removed Adobe Flash or are using Windows 11, here are the [ links to download the latest version of Flash Player ](https://github.com/KuromeSan/FlashPatcher/releases?fbclid=IwAR0MWNV93Mw9WqB2J7sev0-n5Zp_AB8bv2X1nW6b27GVLnF27myO-TVvw24)

To run the Flash Player enabling tool you may :
- compile the C# source code from the repo 
- get the executable from the second link.
- get the executable from attached zip file (I've got it by compiling the code from repo). 

**How to run:**
- Run Flash Pwner.exe 
- Click on `Defuse the bomb`

![Defuse Adobe Flashplayer Bomb](https://user-images.githubusercontent.com/2496596/210278872-12bff01b-7c92-45f3-a35f-6592abbaef76.jpg)

<br /><br />
**Important**: To prevent  Flash Player being disabled again, you need to **disable this service**: `Adobe Flash Player Update Service`
![Dsiable Adobe Flash Player Update Service](https://user-images.githubusercontent.com/2496596/210277785-b198bf8d-ae2f-4b6d-aa47-890b20841ef7.jpg)

I successfully used this tool in 2021 to **re-enable Flash Player** on my computers, and recently (31 Dec. 2022) on a remote machine to be able to use again SlingPlayer 2.0.4. 
- **Log In** , 
- **Setup Assistant**,  
- **TV Guide**, 
- **Favorites**,
- etc. 

**Notes**: 
1. I think that SlingPlayer 2.0.4 still connects to the Sling server, so this may explain why you may need to retry a few times until being able to access some forms (Log In, Setup Assistant & Favorites). 
2. To watch analog from coax I did the setup with SP 1.5.1 but watched with SP 2.0.4.522. which has a much better resolution (1920x540) compared with SP 1.5 which is limited to 640x480.

To get the exe, please see the attached zip:
[DontTouchMyFlash.zip](https://github.com/GerryDazoo/Slinger/files/10333525/DontTouchMyFlash.zip)

**Comment by sp222222** [comment]
yeah, I'm gonna need the exe you mentioned.  I don't know where to start.

**Reply by bradut** [reply]
@sp222222 I have attached a zip file with the executable in my original post above.
